WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift

```diff
@@ -246,25 +246,31 @@ private extension ShippingLabelAddressFormViewController {
                 self.onCompletion(self.viewModel.address)
                 self.navigationController?.popViewController(animated: true)
             case .failure:
-                break
+                self.handleAddressValidationFailure(failure: result.failure)
             }
         }
     }
 }
 
 // MARK: - Utils
 private extension ShippingLabelAddressFormViewController {
-    /// Enqueues the `Apple Maps` Error Notice.
-    ///
-    private func displayAppleMapsErrorNotice() {
-        let notice = Notice(title: Localization.appleMapsErrorNotice, feedbackType: .error, actionTitle: nil, actionHandler: nil)
-        ServiceLocator.noticePresenter.enqueue(notice: notice)
+    private func handleAddressValidationFailure(failure: ShippingLabelAddressFormViewModel.AddressValidationError?) {
+        if let failure = failure {
+            switch failure {
+            case .none:
+                self.displayErrorNotice(title: Localization.addressValidationErrorNotice)
+            case .remote:
+                break
+            }
+        } else {
+            self.displayErrorNotice(title: Localization.addressValidationErrorNotice)
+        }
```

Comment:
I'm not sure if this would be the best place for this function, but I'm also not sure where I could declare it 😅. Do you have thoughts on this @itsmeichigo ? 🤔

Reply:
Thank for so many great tips, it's fixed now!